util/reflect.go

```diff
@@ -8,8 +8,24 @@ import (
 	"github.com/mongodb/grip/message"
 )
 
+func RecursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Value) {
+	// Iterate through each field of the struct.
+	for i := 0; i < structToSet.NumField(); i++ {
```

Comment:
Since this is now public can you add a panic handler at the top which returns an error? That will catch cases if two different types of structs get passed in

Reply:
I added one anyway because panics are stressful, but maybe it's not necessary?